### PR TITLE
Add a lowering config verifier for CPUDoubleTilingExpert.

### PR DESCRIPTION
**Pull request by hanhanW**

### CODE REVIEW

File: iree/compiler/Codegen/LLVMCPU/Passes.cpp [hunk 1]
```diff
@@ -157,6 +157,75 @@ LogicalResult verifyTensorToVectorsPassPipelineConfig(
...
+  // last level of tiling
+  SmallVector<int64_t> nativeVectorSize =
+      loweringConfig.getNativeVectorSizeVals();
+  if (!nativeVectorSize.empty()) {
+    if (nativeVectorSize != loweringConfig.getTileSizeVals(
+                                static_cast<unsigned>(TilingLevel::L1Tiles))) {
+      return op->emitOpError(
+          "native_vector_size must be same as the last level of tiling");
+    }
+  }
```
Comment (hanhanW) [inline iree/compiler/Codegen/LLVMCPU/Passes.cpp:221]: @MaheshRavishankar Are we going to deprecate native_vector_size? It's actually not used. I don't really know if want this check, this is sort of copying from another verifier.

Comment (MaheshRavishankar) [reply iree/compiler/Codegen/LLVMCPU/Passes.cpp:221]: I'd like to deprecate `native_vector_size`. Should I just do it now?

Comment (MaheshRavishankar) [reply iree/compiler/Codegen/LLVMCPU/Passes.cpp:221]: If it is not used, do whatever. Ill clean it up

File: iree/compiler/Codegen/LLVMCPU/Passes.cpp [hunk 2]
```diff
@@ -157,6 +157,75 @@ LogicalResult verifyTensorToVectorsPassPipelineConfig(
...
+    return op->emitOpError("invalid to use 0 in workload_per_wg");
+  }
+
+  if (loweringConfig.getTileSizes().size() != 2) {
+    return op->emitOpError("expected two levels of tile sizes for ")
+           << pipelineName << ", got " << loweringConfig.getTileSizes().size();
+  }
+  SmallVector<int64_t> firstLevelTileSizes = loweringConfig.getTileSizeVals(
+      static_cast<unsigned>(TilingLevel::WorkGroupTiles));
+  if (!firstLevelTileSizes.empty()) {
```
Comment (MaheshRavishankar) [inline iree/compiler/Codegen/LLVMCPU/Passes.cpp:205]: For this part better to follow logic from above. This is unfortunate, but the idea here is that if the configuration is coming in externally we use the `tile_size` to determine the `workgroup_size`. I think this is another thing to clean up. (Ill chat about what is needed here and probably add that to my plate to make us more effective in using the codegen driver). But for the time being, use the same logic from above on those.

**Review by MaheshRavishankar — APPROVED** [review]

File: iree/compiler/Codegen/LLVMCPU/Passes.cpp (hunk 2 above)
Comment (hanhanW) [reply iree/compiler/Codegen/LLVMCPU/Passes.cpp:205]: Got you, done!

**Pull request merged**